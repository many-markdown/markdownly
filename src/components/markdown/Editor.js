import React from 'react';
import PropTypes from 'prop-types';
import styles from './Editor.css';

const Editor = ({ markdown, updateMarkdown }) => (
  <textarea className={styles.Editor} value={markdown} onChange={updateMarkdown} />
);

Editor.propTypes = {
  markdown: PropTypes.string.isRequired,
  updateMarkdown: PropTypes.func.isRequired
};

export default Editor;
