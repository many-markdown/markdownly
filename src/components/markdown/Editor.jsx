import React from 'react';
import styles from './Editor.css';
import { useDispatch } from '../../hooks/appContext';
import { setMarkdown } from '../../actions/markdownActions';
import { useSelector } from '../../hooks/appContext';
import { getMarkdown } from '../../selectors/appSelectors';

const Editor = () => {
  const dispatch = useDispatch();
  const markdown = useSelector(getMarkdown);

  return (
    <textarea className={styles.Editor} value={markdown} onChange={({ target }) => dispatch(setMarkdown(target.value))} />
  );

};
  
export default Editor;
 
