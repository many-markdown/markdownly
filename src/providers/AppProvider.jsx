import React, { useReducer, useEffect } from 'react';
import { AppContext } from '../hooks/appContext';
import reducer, { initialState } from '../reducers/appReducer';
import { setInitialState } from '../actions/markdownActions';

// eslint-disable-next-line react/prop-types
export default function AppProvider({ children }) {

  const [state, dispatch] = useReducer(reducer, initialState);

  useEffect(() => {
    const initialState = localStorage.getItem('FILE_STORAGE'); 
    dispatch(setInitialState(JSON.parse(initialState)));
  }, []);

  return (
    <AppContext.Provider value={{ state, dispatch }} >
      {children}
    </AppContext.Provider>
  );
}
