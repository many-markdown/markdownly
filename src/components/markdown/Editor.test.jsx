import React from 'react';
import { shallow } from 'enzyme';
import Editor from './Editor.jsx';
import AppProvider from '../../providers/AppProvider';

describe('Editor component', () => {
  it('renders Editor', () => {
    const wrapper = shallow(<AppProvider><Editor /></AppProvider>);
    expect(wrapper).toMatchSnapshot();
  });
});
