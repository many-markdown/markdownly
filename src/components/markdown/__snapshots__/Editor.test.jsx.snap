// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor component renders Editor 1`] = `
<ContextProvider
  value={
    Object {
      "dispatch": [Function],
      "state": Object {
        "markdown": "# hello there",
      },
    }
  }
>
  <Editor />
</ContextProvider>
`;
